kube-client/src/config/file_loader.rs

```diff
@@ -89,6 +92,11 @@ impl ConfigLoader {
             .map(|named_user| &named_user.auth_info)
             .ok_or_else(|| KubeconfigError::FindUser(user_name.clone()))?;
 
+        // TODO: docs
+        if cfg!(feature = "rustls-tls") && true && cluster.tls_server_name.is_none() {
+            cluster.tls_server_name = Some("kubernetes.default.svc".to_string());
```

Comment:
For now I moved this workaround to other place (`impl ConfigExt for Config`), so that:
- for `openssl`, tls_server_name is always preserved (but not used for now).
- for `rustls`, tls_server_name is preserved if set in config. Otherwise it is defaulted to `kubernetes.default.svc`.

But I'm not sure If I correctly understood your comment. I'm not familiar with this part of the kube codebase, so maybe it's better to leave workaround logic as a future work?

Reply:
Where you have put it makes the most sense for now. I think I might have gotten mixed up with how this was an override that was instigated by kube itself rather than the user inserting the tls server name into the kubeconfig.

If the the idea is that long term, once rustls gains support for validating ip addresses, we can drop this logic, then this feels like the most appropriate way to propagate it.